modules/cactus-web/src/BrandBar/BrandBar.tsx

```diff
@@ -0,0 +1,192 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled, { createGlobalStyle, css } from 'styled-components'
+
+import { getTopPosition } from '../helpers/positionPopover'
+import { getScrollX } from '../helpers/scrollOffset'
+import { border, boxShadow, fontSize, media, textStyle } from '../helpers/theme'
+
+interface BrandBarProps extends React.HTMLAttributes<HTMLDivElement> {
+  logo?: string | React.ReactElement
+  usernameText: string
+  onProfilePage?: boolean
+}
+
+type ThemeProps = { theme: CactusTheme }
+
+const dropShapeMap = {
+  square: 'border-radius:1px;',
+  intermediate: 'border-radius: 4px;',
+  round: 'border-radius:8px 8px;',
+}
+
+const getDropShape = ({ theme }: ThemeProps) => dropShapeMap[theme.shape]
+
+const MenuButtonStyles = createGlobalStyle`
+  :root {
+    --reach-menu-button: 1;
+  }
+`
+const getDropDownBorder = ({ theme }: ThemeProps) => {
+  if (!theme.boxShadows) {
+    return css`
+      border: ${border(theme, 'lightContrast')};
+      ${getDropShape};
+    `
+  } else {
+    return css(getDropShape)
+  }
+}
+
+const BrandBarBase = (props: BrandBarProps): React.ReactElement => {
+  const { logo, className, usernameText, children } = props
+
+  return (
+    <div className={className}>
+      <LogoWrapper>{typeof logo === 'string' ? <Img alt="Logo" src={logo} /> : logo}</LogoWrapper>
+      <Menu>
+        <MenuButtonStyles />
+        <MenuButton>
+          <DescriptiveProfile mr="8px" />
+          <span>{usernameText}</span>
+          <NavigationChevronDown aria-hidden="true" ml="8px" />
+        </MenuButton>
+        <MenuPopover
+          position={(
+            targetRect,
+            popoverRect
+          ): { minWidth?: number; maxWidth?: number; left?: number; top?: string } => {
+            if (!targetRect || !popoverRect) {
+              return {}
+            }
+
+            const scrollX = getScrollX()
+
+            return {
+              minWidth: targetRect.width,
+              maxWidth: Math.max(targetRect.width, Math.min(targetRect.width * 1.5, 300)),
+              left: targetRect.left + scrollX,
+              ...getTopPosition(targetRect, popoverRect),
+            }
+          }}
+        >
+          <MenuList>{children}</MenuList>
+        </MenuPopover>
+      </Menu>
+    </div>
+  )
+}
+BrandBarBase.Item = MenuItem
+
+export const BrandBar = styled(BrandBarBase)`
+  display: flex;
+  justify-content: center;
+  align-items: flex-end;
+  width: 100%;
+  border-bottom: ${(p): string => border(p.theme, 'lightContrast')};
+  ${(p) =>
+    p.onProfilePage &&
+    `
+    & svg, & button{
+      color: ${p.theme.colors.callToAction}
+    }
+    & button{
+      border-bottom: ${border(p.theme, 'callToAction')};
+    }
+  `}
+  ${(p) => media(p.theme, 'small')} {
+    justify-content: space-between;
+  }
+`
+const LogoWrapper = styled.div`
+  padding: 16px;
+`
+const Img = styled('img')`
+  max-width: 200px;
+  max-height: 80px;
+`
```

Comment:
I think this would be a little more general if they want to use a custom component for the logo.
```suggestion
const LogoWrapper = styled.div`
  padding: 16px;
  max-width: 200px;
  max-height: 80px;
  & > * {
    max-width: 100%;
    max-height: 100%;
  }
`
```